libr/include/r_util/r_annotated_code.h

```diff
@@ -0,0 +1,133 @@
+
+#ifndef R_ANNOTATEDCODE_H
+#define R_ANNOTATEDCODE_H
+
+// #include <r_core.h>
+#include <r_types.h>
+#include <r_vector.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef enum r_syntax_highlight_type_t {
+	R_SYNTAX_HIGHLIGHT_TYPE_KEYWORD,
+	R_SYNTAX_HIGHLIGHT_TYPE_COMMENT,
+	R_SYNTAX_HIGHLIGHT_TYPE_DATATYPE,
+	R_SYNTAX_HIGHLIGHT_TYPE_FUNCTION_NAME,
+	R_SYNTAX_HIGHLIGHT_TYPE_FUNCTION_PARAMETER,
+	R_SYNTAX_HIGHLIGHT_TYPE_LOCAL_VARIABLE,
+	R_SYNTAX_HIGHLIGHT_TYPE_CONSTANT_VARIABLE,
+	R_SYNTAX_HIGHLIGHT_TYPE_GLOBAL_VARIABLE,
+} RSyntaxHighlightType;
+
+/**
+ * enum r_code_annotation_type_t - typedefed as RCodeAnnotationType and this gives types of annotation
+ *
+ * There are two kinds of RCodeAnnotation. One for offset, which of the type 
+ * R_CODE_ANNOTATION_TYPE_OFFSET and other one is for syntax highlight, which is
+ * of the type R_CODE_ANNOTATION_TYPE_SYNTAX_HIGHLIGHT.
+ * R_CODE_ANNOTATION_TYPE_OFFSET is for representing annotations that gives an offset for 
+ * a range while R_CODE_ANNOTATION_TYPE_SYNTAX_HIGHLIGHT is for representing the
+ * kind of data the range represents. Here, range refers to the range of annotation.
+ */
+
+typedef enum r_code_annotation_type_t {
+	R_CODE_ANNOTATION_TYPE_OFFSET,
+	R_CODE_ANNOTATION_TYPE_SYNTAX_HIGHLIGHT,
+	// ...
+} RCodeAnnotationType;
+
+typedef struct r_code_annotation_t {
+	size_t start;
+	size_t end;
+	RCodeAnnotationType type;
+	union {
+		struct {
+			ut64 offset;
+		} offset;
+
+		struct {
+			RSyntaxHighlightType type;
+		} syntax_highlight;
+	};
+} RCodeAnnotation;
+
+typedef struct r_annotated_code_t {
+	char *code; // owned
+	RVector /*<RCodeAnnotation>*/ annotations;
+} RAnnotatedCode;
+
+/**
+ * r_annotated_code_new() - Creates a new RAnnotatedCode structure and returns it's pointer.
+ * @code: Literal code for which the RAnnotatedCode structure will be created .
+ *
+ * This functions creates a new RAnnotatedCode structure.
+ * RAnnotatedCode.code will be initialized as the character array passed.
+ * Here, code must be a string that can deallocated.
+ * This will initialize RVector /*<RCodeAnnotation>*\/ annotations as well.
+ * 
+ * Return: Pointer to the new RAnnotatedCode structure created.
```

Comment:
```suggestion
 * r_annotated_code_new() - Creates a new RAnnotatedCode structure and returns its pointer.
 * @code: Literal code for which the RAnnotatedCode structure will be created .
 *
 * This functions creates a new RAnnotatedCode structure.
 * RAnnotatedCode.code will be initialized as the character array passed.
 * Here, code must be a string that can deallocated.
 * This will initialize RVector /*<RCodeAnnotation>*\/ annotations as well.
 * 
 * Return: Pointer to the new RAnnotatedCode structure created.
```

Reply:
Changed as suggested.